editor/src/clj/editor/engine.clj

```diff
@@ -251,18 +256,21 @@
       (copy-dmengine-dependencies! engine-dir extender-platform)
       engine-file)))
 
-(defn launch! [^File engine project-directory prefs debug?]
+(defn launch! [^File engine project-directory prefs debug? instance-index]
   (let [defold-log-dir (some-> (System/getProperty "defold.log.dir")
                                (File.)
                                (.getAbsolutePath))
         command (.getAbsolutePath engine)
         args (cond-> []
-               defold-log-dir
-               (into ["--config=project.write_log=1"
-                      (format "--config=project.log_dir=%s" defold-log-dir)])
+                     defold-log-dir
+                     (into ["--config=project.write_log=1"
+                            (format "--config=project.log_dir=%s" defold-log-dir)])
+
+                     debug?
+                     (into ["--config=bootstrap.debug_init_script=/_defold/debugger/start.luac"])
 
-               debug?
-               (into ["--config=bootstrap.debug_init_script=/_defold/debugger/start.luac"]))
+                     (> instance-index 0)
+                     (into [(format "--config=project.instance_index=%d" instance-index)]))
```

Comment:
I wonder if maybe we should set the instance_index as a process environment variable instead of a project property? Like DM_SERVICE_PORT and DM_QUIT_ON_ESC? Perhaps DM_ENGINE_INSTANCE_INDEX?

At runtime it can probably be read using `os.getenv("DM_ENGINE_INSTANCE_INDEX")` or if we expose `dmSys::GetEnv()` to Lua?

Reply:
Yes, we can. But is it somehow better?

Reply:
I think it is. Entries in game.project are mostly user-defined right? Do we have other example of where we inject values into project settings on startup? In the case of the instance count it is some extra data which the editor defines and provides on startup, much like the service port, which is why I'm thinking that an env variable makes more sense.

Reply:
> Do we have other example of where we inject values into project settings on startup?

yes, we use it in a few proaces: 
https://github.com/defold/defold/blob/8317584212f3047dd6881f38ba6f4416dcf16bb1/editor/src/clj/editor/engine.clj#L90

https://github.com/defold/defold/blob/8317584212f3047dd6881f38ba6f4416dcf16bb1/editor/src/clj/editor/engine.clj#L92

https://github.com/defold/defold/blob/8317584212f3047dd6881f38ba6f4416dcf16bb1/editor/src/clj/editor/engine.clj#L262

https://github.com/defold/defold/blob/8317584212f3047dd6881f38ba6f4416dcf16bb1/editor/src/clj/editor/engine.clj#L265


I don't have a strong opinion here. If we want - I can rewrite it. It will be a bit more Lua code to get default value 0, but it's not critical.

Reply:
Ah, ok, if there is a precedent, then let us keep it as it is now!